File: ios/CreatePrivateDealerOS/CreatePrivateDealerOS/Core/Utils/PhoneMasker.swift
import Foundation

enum PhoneMasker {
    static func mask(_ phone: String, revealFull: Bool) -> String {
        if revealFull {
            return phone
        }

        let digits = phone.filter(\.isNumber)
        guard digits.count >= 4 else {
            return "****"
        }

        let suffix = String(digits.suffix(4))
        return "***-***-\(suffix)"
    }
}

File: flutter/create_private_dealer_os/lib/core/utils/phone_masker.dart
class PhoneMasker {
  const PhoneMasker._();

  static String mask(String phone, {required bool revealFull}) {
    if (revealFull) {
      return phone;
    }

    if (phone.isEmpty) {
      return 'No phone';
    }

    if (phone.length <= 4) {
      return '****';
    }

    final String suffix = phone.substring(phone.length - 4);
    return '***-***-$suffix';
  }
}
